### PR TITLE
Geom density2d

### PR DESCRIPTION
**Pull request by maxwellfundi**
@africanmathsinitiative/developers  this is ready for review. 

Fixes #4220

### CODE REVIEW

**Comment by maxwellfundi** [comment]
@rdstern  we have got the geom_density2d added and working. 

image

I have a few questions though, from the look of things the **stat** parameter we got just **density2d** and for **position** parameter we got **identity**. Do you know if we could have more than these? 

Secondly, the parameters n- which number of grid points in each direction and h which is the Bandwidth, as indicated from the help does not have default values even when i looked at the geom code. Whats your opinion for a suitable defaults for these?

**Comment by Muthenya** [comment]
This looks good.

**Comment by maxwellfundi** [comment]
great then @dannyparsons  can look at it

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -441,24 +442,32 @@ Public Class ucrGeom
...
+        'not sure of other stats and positons here
+        clsgeom_density2d.AddLayerParameter("stat", "list", Chr(34) & "density2d" & Chr(34), lstParameterStrings:={Chr(34) & "density2d" & Chr(34)})
+        clsgeom_density2d.AddLayerParameter("position", "list", Chr(34) & "identity" & Chr(34), lstParameterStrings:={Chr(34) & "identity" & Chr(34)})
+        clsgeom_density2d.AddLayerParameter("lineend", "list", Chr(34) & "butt" & Chr(34), lstParameterStrings:={Chr(34) & "round" & Chr(34), Chr(34) & "butt" & Chr(34), Chr(34) & "square" & Chr(34)})
+        clsgeom_density2d.AddLayerParameter("linejoin", "list", Chr(34) & "round" & Chr(34), lstParameterStrings:={Chr(34) & "round" & Chr(34), Chr(34) & "mitre" & Chr(34), Chr(34) & "bevel" & Chr(34)})
+        clsgeom_density2d.AddLayerParameter("na.rm", "boolean", "FALSE")
+        clsgeom_density2d.AddLayerParameter("show.legend", "list", "NA", lstParameterStrings:={"NA", "TRUE", "FALSE"})
+        clsgeom_density2d.AddLayerParameter("inherit.aes", "boolean", "TRUE")
+        clsgeom_density2d.AddLayerParameter("contour", "boolean", "FALSE")
```
Comment (dannyparsons) [inline instat/ucrGeom.vb]: I've just noticed that the default for contour is TRUE not FALSE.

**Pull request merged**